doc/aspice_swe3_4/example/iceoryx_component/include/example_component/example_derived_class.hpp

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#ifndef IOX_DOC_EXAMPLE_COMPONENT_EXAMPLE_DERIVED_CLASS_HPP
+#define IOX_DOC_EXAMPLE_COMPONENT_EXAMPLE_DERIVED_CLASS_HPP
+
+#include "source/example_base_class.hpp"
+
+
+namespace example
+{
+/// @brief Short description
+/// @code
+///   minimalistic working example which uses all public methods
+/// @endcode
+/// @swcomponent example_component
+void someFreeFunction();
+
+/// @brief Forward declaration of other class
+class SomeOtherClass;
```

Comment:
Why Other? Is there another one? SomeClass?

Reply:
Its just an example to show that also forward declarations should be documented. I will make that more clear in the next PR